### PR TITLE
[LLC] Implement Short LRO

### PR DESCRIPTION
**Pull request by ShivangiReja**
Fixes: https://github.com/Azure/autorest.csharp/issues/1684

As per https://gist.github.com/ShivangiReja/7949773c9759f9a597d21787afd02302, this PR adds the following implementation: 

* If a flag `x-ms-long-running-reallyLong: true` in `x-ms-long-running-operation-options` has set, then we will implement Long LRO, otherwise we will implement Short LRO
* Add "Start" prefix for Long LRO methods
* Add waitForCompletion = false parameter for Start[Method] (Long LRO) 
* Add waitForCompletion = true parameter for [Method] (Short LRO)
* Make SLRO default for LLC
* Add test project to cover Short LRO, Long LRO, Pageable Short LRO and Pageable Long LRO

### CODE REVIEW

File: test/TestProjects/LRO-LowLevel/LRO-LowLevel.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ResourceListResult"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location",
+          "x-ms-long-running-reallyLong": true
```
Comment (ShivangiReja) [inline test/TestProjects/LRO-LowLevel/LRO-LowLevel.json:100]: Please update `"x-ms-long-running-reallyLong"` flag name as per the Arch board final decision. Here are some suggestions -  `"x-ms-long-running-long"` or `"x-ms-long-running-always"`(suggested by @tg-msft)

File: src/AutoRest.CSharp/AutoRest.CSharp.csproj [hunk 2]
```diff
@@ -17,8 +17,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Azure.Core" Version="1.21.0-alpha.20211029.1" />
-    <PackageReference Include="Azure.Core.Experimental" Version="0.1.0-preview.16" />
+    <PackageReference Include="Azure.Core" Version="1.22.0-alpha.20211119.1" />
+    <PackageReference Include="Azure.Core.Experimental" Version="0.1.0-preview.18" />
```
Comment (ShivangiReja) [inline src/AutoRest.CSharp/AutoRest.CSharp.csproj:21]: Sync version of WaitForCompletion method was not existed in the older Azure.Core version(`"1.21.0-alpha.20211029.1"`), so I had to update the core version in order to call `operation.WaitForCompletion();` for LLC LRO methods. [Here](https://github.com/Azure/autorest.csharp/blob/3c8aa4e060ca4729082a78bbd76ab43ade5facb2/test/TestProjects/LRO-LowLevel/Generated/LROServiceClient.cs#L112) is an example.

File: src/assets/Generator.Shared/LowLevelFuncOperation.cs [hunk 3]
```diff
@@ -44,6 +44,12 @@ internal LowLevelFuncOperation(ClientDiagnostics clientDiagnostics, HttpPipeline
         /// <inheritdoc />
         public override ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) => _operation.UpdateStatusAsync(cancellationToken);
 
+        /// <inheritdoc />
+        public override Response<T> WaitForCompletion(CancellationToken cancellationToken = default) => _operation.WaitForCompletion(cancellationToken);
+
+        /// <inheritdoc />
+        public override Response<T> WaitForCompletion(TimeSpan pollingInterval, CancellationToken cancellationToken = default) => _operation.WaitForCompletion(pollingInterval, cancellationToken);
+
```
Comment (ShivangiReja) [inline src/assets/Generator.Shared/LowLevelFuncOperation.cs:52]: Sync version of WaitForCompletion method was not existed before. It is recently added in this [PR](https://github.com/Azure/azure-sdk-for-net/pull/22420/files#diff-d8fef9e5888142bd6d9a5398c2d61a54cde4d38e79fdd50a204650a1cf44abaaR139). It is needed in order to call `operation.WaitForCompletion();` for LLC LRO methods. [Here](https://github.com/Azure/autorest.csharp/blob/3c8aa4e060ca4729082a78bbd76ab43ade5facb2/test/TestProjects/LRO-LowLevel/Generated/LROServiceClient.cs#L112) is an example.

File: test/TestProjects/LRO-LowLevel/Generated/LROServiceClient.cs [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace LRO_LowLevel
+{
+    /// <summary> The LROService service client. </summary>
+    public partial class LROServiceClient
```
Comment (ShivangiReja) [inline test/TestProjects/LRO-LowLevel/Generated/LROServiceClient.cs:20]: Here is the Generated test project which covers Short LRO, Long LRO, Pageable Short LRO, Pageable Long LRO

File: test/AutoRest.TestServerLowLevel.Tests/LRO_LowLevelTests.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            Assert.NotNull(waitForCompletionParam);
+            Assert.AreEqual(waitForCompletionParam.ParameterType, typeof(bool));
+            Assert.AreEqual(waitForCompletionParam.IsOptional, true);
+            Assert.AreEqual(waitForCompletionParam.DefaultValue, true);
+        }
+
+        [TestCase("StartDoSomethingLROAsync")]
+        [TestCase("StartDoSomethingLRO")]
+        [TestCase("StartPageableLROAsync")]
+        [TestCase("StartPageableLRO")]
```
Comment (ShivangiReja) [inline test/AutoRest.TestServerLowLevel.Tests/LRO_LowLevelTests.cs:32]: Here are the unit tests to verify Short LRO, Long LRO, Pageable Short LRO, Pageable Long LRO methods.

File: test/TestProjects/LRO-LowLevel/LRO-LowLevel.json [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Resource"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location",
+          "x-ms-long-running-reallyLong": true
```
Comment (ArthurMa1978) [inline test/TestProjects/LRO-LowLevel/LRO-LowLevel.json:47]: This option has not defined in https://github.com/stankovski/AutoRest/blob/master/Documentation/swagger-extensions.md, and never be used by a real service (no record in https://github.com/Azure/azure-rest-api-specs).
There also had a similar PR merged in Aug. 20 https://github.com/Azure/autorest.csharp/pull/1473 that you made.
I talked with swagger team and they have no idea about this feature, so would like give more information about this change? And to this kind of feature it definitely should be written down in our swagger spec.
Loop @lirenhe  and @akning-ms .

Comment (akning-ms) [reply test/TestProjects/LRO-LowLevel/LRO-LowLevel.json:47]: Not sure whether @johanste @lmazuel @allenjzhang  whether know this change/feature for existing LRO swagger extension?

**Comment by ArthurMa1978** [comment]
According to https://github.com/Azure/azure-sdk/issues/3662#issuecomment-985112894 we shouldn't have one LRO pattern

**Comment by AlexanderSher** [comment]
Closing due to the changed design

**Pull request closed without merging**